punchbowl/data/tests/test_wcs.py

```diff
@@ -80,115 +80,246 @@ def test_extract_crota_helio():
     wcs_helio.wcs.pc = pc
     assert np.allclose(extract_crota_from_wcs(wcs_helio), 10 * u.deg)
 
-def test_wcs_many_point_2d_check():
-    m = NormalizedMetadata.load_template("CTM", "2")
-    date_obs = Time("2024-01-01T00:00:00", format='isot', scale='utc')
-    m['DATE-OBS'] = str(date_obs)
+def test_helio_to_celestial_wcs_many_points():
+    for date_obs, crval1, crval2, crot in product(
+            ["2021-03-20T00:00:00", "2021-01-20T00:00:00"],
+            [-20, 10, 30],
+            [-10, 0, 10],
+            [-30, 0, 30] * u.deg):
+        earth = get_earth(date_obs)
+        wcs_helio = WCS({"CRVAL1": crval1,
+                         "CRVAL2": crval2,
+                         "CRPIX1": 2047.5,
+                         "CRPIX2": 2047.5,
+                         "NAXIS1": 4096,
+                         "NAXIS2": 4096,
+                         "CDELT1": 0.0225,
+                         "CDELT2": 0.0225,
+                         "CUNIT1": "deg",
+                         "CUNIT2": "deg",
+                         "CTYPE1": "HPLN-AZP",
+                         "CTYPE2": "HPLT-AZP",
+                         "PC1_1": np.cos(crot).value,
+                         "PC1_2": -np.sin(crot).value,
+                         "PC2_1": np.sin(crot).value,
+                         "PC2_2": np.cos(crot).value,
+                         "DATE-OBS": date_obs,
+                         "MJD-OBS": Time(date_obs).mjd,
+                         "DSUN_OBS": earth.radius.to_value(u.m),
+                         "HGLN_OBS": earth.lon.to_value(u.deg),
+                         "HGLT_OBS": earth.lat.to_value(u.deg),
+                         "PV2_1": 0,
+                         })
 
-    sun_radec = get_sun(date_obs)
-    wcs_celestial = WCS({"CRVAL1": sun_radec.ra.to(u.deg).value,
-                         "CRVAL2": sun_radec.dec.to(u.deg).value,
+        wcs_celestial = calculate_celestial_wcs_from_helio(wcs_helio)
+
+        npoints = 20
+        xs, ys = np.meshgrid(
+            np.linspace(0, wcs_helio.pixel_shape[0], npoints),
+            np.linspace(0, wcs_helio.pixel_shape[1], npoints),
+        )
+
+        hp_coords = wcs_helio.pixel_to_world(xs, ys)
+        # These come out claiming to be ICRS, but they're really GCRS (since it doesn't seem possible to tell a WCS
+        # that it's GCRS)
+        eq_coords = wcs_celestial.pixel_to_world(xs, ys)
+        hp_to_eq_coords = hp_coords.transform_to('gcrs')
+
+        eq_ra = eq_coords.ra.to_value(u.deg)
+        hp_to_eq_ra = hp_to_eq_coords.ra.to_value(u.deg)
+        straddles_wrap = np.nonzero(np.abs(eq_ra - hp_to_eq_ra) > 350)
+        for r, c in zip(*straddles_wrap):
+            if eq_ra[r, c] > 350:
+                eq_ra[r, c] -= 360
+            else:
+                hp_to_eq_ra[r, c] -= 360
+
+        np.testing.assert_allclose(eq_ra, hp_to_eq_ra, atol=1e-12)
```

Comment:
These are good lines! The coordinates computed from the equatorial WCS and those compute from the HP WCS and converted to equatorial agree to within 1e-12 degrees!